dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/event/EnrollmentController.java

```diff
@@ -326,6 +338,27 @@ public void deleteEnrollment( @PathVariable String id, HttpServletRequest reques
         webMessageService.send( WebMessageUtils.importSummary( importSummary ), response, request );
     }
 
+    // -------------------------------------------------------------------------
+    // QUEUED IMPORT
+    // -------------------------------------------------------------------------
+
+    @RequestMapping( value = "/queue", method = RequestMethod.POST, consumes = "application/json" )
+    public void postQueuedJsonEvents( @RequestParam( defaultValue = "CREATE_AND_UPDATE" ) ImportStrategy strategy,
+        HttpServletResponse response, HttpServletRequest request, ImportOptions importOptions ) throws Exception
```

Comment:
Should we throw WebMessageException instead of Exception?

Reply:
Used Exception because we also use getInputStream (which can throw IOException), changed to WebMessageException + IOException.